pkg/dao/environments.go

```diff
@@ -0,0 +1,362 @@
+package dao
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/content-services/content-sources-backend/pkg/api"
+	ce "github.com/content-services/content-sources-backend/pkg/errors"
+	"github.com/content-services/content-sources-backend/pkg/models"
+	"github.com/content-services/yummy/pkg/yum"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+type environmentDaoImpl struct {
+	db *gorm.DB
+}
+
+func GetEnvironmentDao(db *gorm.DB) EnvironmentDao {
+	// Return DAO instance
+	return environmentDaoImpl{
+		db: db,
+	}
+}
+
+func (r environmentDaoImpl) isOwnedRepository(orgID string, repositoryConfigUUID string) (bool, error) {
```

Comment:
I think if you changed your function parameters to be `isOwnedRepository(db *gorm.DB, orgID string, repositoryConfigUUID string)`, rather than make this a method of environmentDaoImpl, you could share this function between all 3 content types.

Reply:
ah right, i haven't done that yet. will add that in too

Reply:
added this, thanks for the tip!